enterprise/payara.tooling/test/unit/src/org/netbeans/modules/payara/tooling/data/JDKVersionTest.java

```diff
@@ -40,23 +40,23 @@
     public void parseJDKVersion() {
         Map<String, JDKVersion> jdkVersions = new HashMap<>();
         jdkVersions.put("1.8",
-                new JDKVersion((short) 1, Optional.of((short) 8), Optional.of((short) 0), Optional.of((short) 0), Optional.empty()));
+                new JDKVersion((short) 1, Optional.of((short) 8), Optional.of((short) 0), Optional.of((short) 0), Optional.empty(), Optional.empty()));
```

Comment:
Unrelated: I don't like `Optional` and these lines show why: wouldn't it be simpler, to just use `null` instead of `Optional.empty()` and `(short)0` instead of `Optional.of(...)`?

Reply:
Well, I like `Optional` when it's used as intended. I agree with you here!

Reply:
This change will require refactoring many functions and due to signature issues deprecated functions need to be added, I will look into this in separate PR.